actionview/lib/action_view/template/renderable.rb

```diff
@@ -4,20 +4,31 @@ module ActionView
   class Template
     # = Action View Renderable Template for objects that respond to #render_in
     class Renderable # :nodoc:
-      def initialize(renderable)
+      def initialize(renderable, formats, &block)
         @renderable = renderable
+        @formats = formats
+        @block = block
       end
 
       def identifier
         @renderable.class.name
       end
 
-      def render(context, *args)
-        @renderable.render_in(context)
+      def render(context, locals)
+        case @renderable.method(:render_in).arity
+        when 1
+          ActionView.deprecator.warn <<~WARN
+            #render_in signatures without options are deprecated.
```

Comment:
@rafaelfranca @joelhawksley I've added a deprecation here along with test coverage. I struggled to convey a more detailed deprecation message. If you're able, please share any suggestions you have.